resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -69,6 +70,10 @@ private[spark] class Client(
 
   private val isClusterMode = sparkConf.get("spark.submit.deployMode", "client") == "cluster"
 
+  private val isClientUnmanagedAMEnabled =
+    sparkConf.getBoolean("spark.yarn.un-managed-am", false) && !isClusterMode
```

Comment:
This should be a config constant. Also `unmanagedAM` is more in line with other config names.

Reply:
Updated the config name and also added config constants.